src/Protocol/DocumentUri.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.IO;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace OmniSharp.Extensions.LanguageServer.Protocol
+{
+    /// <summary>
+    ///     Helper methods for working with LSP document URIs.
+    /// </summary>
+    public static class DocumentUri
```

Comment:
This is an interesting class. Would it handle VS Code's conflicting `Uri` type (aka the fact that `C:` is encoded to `C%3A`)

Reply:
We probably should add that here with some unit tests to make sure.

This was brought up from the client I think.

Reply:
In those tests, can you also include tests of Chinese characters or Cyrillic or something? Those don't always play nicely with the `Uri` type...